From 9e04eb0d5fc07617d5e37df991eac11d5812c92e Mon Sep 17 00:00:00 2001
From: Perry Yuan <perry_yuan@dell.com>
Date: Sun, 7 Mar 2021 21:50:49 -0500
Subject: [PATCH] hwdb: 60-keyboard:: Update Dell Privacy Micmute Hotkey Map
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1926547
Origin: upstream, https://github.com/systemd/systemd/commit/9e04eb0d5fc
Origin: upstream, https://github.com/systemd/systemd/commit/5ed0ea29287

Dell new Privacy feature provide new hardware level privacy
protect for users
This patch remaps scancode 0x120001 to key code F20 micmute
The old matching string cannot cover some other Dell products
which have the privacy feature,expand the string to all the system
that can load the privacy driver,privacy driver already detect the
system if it can support this feature. So here we can safely just
map the micmute key to scancode 0x120001

Signed-off-by: Perry Yuan <perry_yuan@dell.com>
---
 hwdb.d/60-keyboard.hwdb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -373,6 +373,10 @@ evdev:name:Dell WMI hotkeys:dmi:bvn*:bvr
 evdev:name:Dell WMI hotkeys:dmi:bvn*:bvr*:bd*:svnDell*:pnPrecision*
  KEYBOARD_KEY_100150=f20                                   # Mic mute toggle, should be micmute
 
+# Dell Latitude privacy microphone mute
+evdev:name:Dell Privacy Driver:dmi:bvn*:bvr*:bd*:svnDell*:pnLatitude*:*
+ KEYBOARD_KEY_120001=f20                                # Mic mute toggle, should be micmute
+
 ###########################################################
 # Everex
 ###########################################################
